python/lsst/pipe/base/pipelineIR.py

```diff
@@ -195,6 +240,27 @@ def to_primitives(self) -> dict:
         accumulate.update(self.rest)
         return accumulate
 
+    def formatted(self, parameters: ParametersIR) -> ConfigIR:
+        """Returns a new ConfigIR object that is formatted according to the
+        specified parameters
+
+        Parameters
+        ----------
+        parameters : ParametersIR
+            Object that contains variable mappings used in substitution.
+
+        Returns
+        -------
+        config : ConfigIR
+            A new ConfigIR object formatted with the input parameters
+        """
+        new_config = copy.deepcopy(self)
+        for key, value in new_config.rest.items():
+            match = re.match("parameters[.](.*)", value)
+            if match and match.group(1) in parameters:
+                new_config.rest[key] = parameters[match.group(1)]
```

Comment:
Should this warn or raise `if match and match.group(1) not in parameters`?  Or will that fail later in the pipeline construction process?

Reply:
I left this open so that if there is a parameter in the pipeline and it matches it is used. But if someone had a value that contained this string that was not a parameter it would allow it to continue in case there was a task somewhere that needed this as a string or something for some other reason. I didn't want to be any more restrictive than I needed to be. Do you think it should be some other way (I definitely am open to other viewpoints here, I just went with what seemed best to me)

Reply:
I went back and forth, and I think I've settled on "a warning is good."  In the case where a parameter wasn't defined right, it can at least say "No parameter named 'typoe' found".  In the case where someone defines a `goodIdea: config.parameters.badIdea`, the warning will hopefully annoy them into not using that name.